### PR TITLE
Remove input_quant_func from AffineQuantizedTensor subclass

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
Currently we have a input_quant_func in the AffineQuantizedTensor, which is a bit convoluted, we want to use a separate LinearActAffineQuantizedTensor subclass for activation quantization (dynamic quantization) instead

Test Plan:
python test/quantization/test_quant_api.py -k test_quantized_tensor_subclass_8da4w

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torchao/quantization/subclass.py [hunk 1]
```diff
@@ -36,6 +38,30 @@
...
 
+def _aqt_is_int8(aqt):
+    """Check if an AffineQuantizedTensor is int8 quantized Tensor"""
+    return (
+        aqt.int_data.dtype == torch.int8 and
+        aqt.quant_min is None or aqt.quant_min == -128 and
+        aqt.quant_max is None or aqt.quant_max == 127
+    )
+
+def _aqt_is_int8_reduce_range(aqt):
```
Comment (cpuhrsch) [inline torchao/quantization/subclass.py:49]: So an aqt can be both int8 and int8 reduced_range if quant_min is None.

Should we require quant_min to always be defined? Is there an implied value when it's None?

Comment (cpuhrsch) [reply torchao/quantization/subclass.py:49]: Also nit I'd use `reduced` instead of `reduce`, just because `reduce` is a verb. This doesn't matter quite that much since it's a private API for now.

File: torchao/quantization/subclass.py [hunk 2]
```diff
@@ -805,18 +861,18 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
...
-
+                if isinstance(input_tensor, AffineQuantizedTensor):
+                    input_tensor = input_tensor.dequantize()
+                return torch.nn.functional.linear(input_tensor, weight_tensor, bias)
         try:
             with torch._C.DisableTorchFunctionSubclass():
                 return func(*args, **kwargs)
         except:
-            print(f"ERR: subclass doesn't implement {func}")
+            print(f"ERR: AffineQuantizedTensor subclass doesn't implement {func}")
```
Comment (cpuhrsch) [inline torchao/quantization/subclass.py:875]: A blanked `except` here will say that `func` is not implemented when in fact `func` might have raised a valid `ValueError` or such when an argument was chosen incorrectly. This error should also now be raised from within torch dispatch itself

File: torchao/quantization/subclass.py (hunk 1 above)
Comment (jerryzh168) [reply torchao/quantization/subclass.py:49]: oh sorry, I think we want to remove `None` here

**Review by cpuhrsch — APPROVED** [review]
Great :) Let's move AffineQuantizedTensor into dtypes next and create a PyTorch style conversion function? We should also not need to use torch_function to overwrite `linear`, but it makes sense to do it as a follow up because it'll require us to add support for detach, view, addmm, etc. to `AffineQuantizedTensor`

File: torchao/quantization/subclass.py (hunk 2 above)
Comment (jerryzh168) [reply torchao/quantization/subclass.py:875]: Sure I can change the error message. 
you mean we want to reraise the error here?

**Comment by jerryzh168** [comment]
> Great :) Let's move AffineQuantizedTensor into dtypes next and create a PyTorch style conversion function? We should also not need to use torch_function to overwrite `linear`, but it makes sense to do it as a follow up because it'll require us to add support for detach, view, addmm, etc. to `AffineQuantizedTensor`

sounds good. main thing is transpose, we need to think about how to support that with the scales/zero_point and block_size arg

File: torchao/quantization/subclass.py (hunk 2 above)
Comment (jerryzh168) [reply torchao/quantization/subclass.py:875]: followed up offline, will remove try except here
